GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts

```diff
@@ -1,104 +1,296 @@
+
+    constructor(howl: Howl) {
+      this._howl = howl;
     }
 
-    // Redefine `stop`/`play`/`pause` to ensure the status of the sound
-    // is immediately updated (so that calling `stopped` just after
-    // `play` will return false).
-    stop() {
-      this._paused = false;
-      this._stopped = true;
-      return super.stop();
+    /**
+     * Returns true if the associated howl is fully loaded.
+     */
+    isLoaded(): boolean {
+      return this._howl.state() === 'loaded';
     }
 
-    play() {
-      this._paused = false;
-      this._stopped = false;
-      return super.play();
+    /**
+     * Begins playback of the sound, or if the Howl is still loading, schedule playing for once it loads.
+     * @returns The current instance for chaining.
+     */
+    play(): this {
+      if (this.isLoaded()) {
+        const newID = this._howl.play(this._id === null ? undefined : this._id);
+        this._id = newID;
+
+        // Manually handle the play event before we have an ID.
+        // Before loading, howler won't register events as without an ID we cannot set a listener.
+        // Once we have an ID, we can transfer control of the events to howler.
+        // We also need to call them once as Howler doesn't for the first play event.
+        this._onPlay.forEach((func) => {
+          // Transfer the event to howler now that we have an ID
+          this.on('play', func);
+          func(newID);
+        });
+        this._oncePlay.forEach((func) => func(newID));
+        this._onPlay = [];
+        this._oncePlay = [];
+      } else this._howl.once('load', () => this.play()); // Play only once the howl is fully loaded
+
+      return this;
     }
 
-    pause() {
-      this._paused = true;
-      this._stopped = false;
-      return super.pause();
+    /**
+     * Pauses playback of the sound, saving the seek of playback.
+     * @returns The current instance for chaining.
+     */
+    pause(): this {
+      if (typeof this._id !== 'undefined') this._howl.pause(this._id);
+      return this;
+    }
+
+    /**
+     * Stops playback of the sound, resetting seek to 0.
+     * @returns The current instance for chaining.
+     */
+    stop(): this {
+      if (typeof this._id !== 'undefined') this._howl.stop(this._id);
+      return this;
+    }
+
+    /**
+     * Check if the sound is currently playing.
+     */
+    playing(): boolean {
+      return (
+        (this._id !== null ? this._howl.playing(this._id) : false) ||
+        !this.isLoaded() // Loading is considered playing
+      );
+    }
+
+    /**
+     * Check if the sound is currently paused.
+     */
+    paused(): boolean {
+      return !this.playing();
+    }
+
+    /**
+     * Check if the sound is currently stopped.
+     */
+    stopped(): boolean {
+      return this.paused() && this.getSeek() === 0;
+    }
+
+    /**
+     * Get the sound playback rate.
+     */
+    getRate(): float {
+      if (typeof this._id === 'undefined') return 0;
```

Comment:
nitpicking: I recommend to avoid typeof in almost all cases. A simple `this._id === undefined` will work.
Or better, `this._id === null` (undefined is usually a problem in JavaScript, while null is a way to say "I made this on purpose"). 

In both cases, be sure to assign it in the constructor, to avoid the JS engine to be forced to dynamically add a field to the class at runtime: `this._id = undefined` or `this._id = null` in the constructor.